pkg/compute/deploy.go

```diff
@@ -92,34 +83,58 @@ func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
 			return fmt.Errorf("error listing service versions: %w", err)
 		}
 
-		version, err := getLatestIdealVersion(versions)
+		version, err = getLatestIdealVersion(versions)
 		if err != nil {
 			return fmt.Errorf("error finding latest service version")
 		}
 
-		if version.Active || version.Locked {
-			progress.Step("Cloning latest version...")
-			version, err = c.Globals.Client.CloneVersion(&fastly.CloneVersionInput{
-				Service: serviceID,
-				Version: version.Number,
-			})
-			if err != nil {
-				return fmt.Errorf("error cloning latest service version: %w", err)
-			}
+		c.version = version.Number
+	}
+
+	progress.Step("Validating package...")
+
+	if err := validate(c.path); err != nil {
+		return err
+	}
+
+	// Compare local package hashsum against existing service package version
+	// and exit early with message if identical.
+	if p, err := c.Globals.Client.GetPackage(&fastly.GetPackageInput{
+		Service: serviceID,
+		Version: c.version,
+	}); err == nil {
+		hashSum, err := getHashSum(c.path)
+		if err != nil {
+			return fmt.Errorf("error getting package hashsum: %w", err)
+		}
+
+		if hashSum == p.Metadata.HashSum {
+			progress.Done()
+			text.Info(out, "Skipping package deployment, local and service version hashsums are identical. (service %v, version %v) ", serviceID, c.version)
```

Comment:
Perhaps the customer doesn't care how we're determining that the local and remote packages are the same. ie. We can skip the bit about the "hashsum?" Seems fine either way, though.

How likely is it that the package will be the same?

Reply:
:+1: Have removed the "hashsum" portion of the message.

> How likely is it that the package will be the same?

Not that likely, however this ensures running `fastly compute deploy` twice is idempotent and at the very least should hint to users that they may have forgotten to  run `fastly compute build` after changing some code.